raster/src/main/scala/geotrellis/raster/ArrayMultiBandTile.scala

```diff
@@ -269,4 +269,20 @@ class ArrayMultiBandTile(bands: Array[Tile]) extends MultiBandTile {
     }
     result
   }
+
+  def subset(bands: Seq[Int]): ArrayMultiBandTile = {
+    val newBands = Array.ofDim[Tile](bands.size)
+    var i = 0
+
+    assert(bands.size <= this.bandCount)
```

Comment:
this should use `require`, that produces `IllegalArgumentException` and gives a chance for an error message.